kaybee/internal/conf/conf.go

```diff
@@ -5,120 +5,105 @@
 // and how to merge statements according to one or more strategies
 package conf
 
-import "strings"
-
-// "log"
-// "os"
-// "github.com/sap/project-kb/kaybee/internal/errors"
-// homedir "github.com/mitchellh/go-homedir"
-
-// "github.com/spf13/viper"
-// "regexp"
-
-// A Configuration encapsulates the relevant fields
-// of configurations objects. This is used to hide the different configuration schema
-// versions an provide a general API. This is the type that is ready everywere in the
-// other packages.
-type Configuration struct {
-	version        string
-	backend        string
-	exportDenylist []string
-	sources        []Source
-	policies       []Policy
-	export         []ExportScript
-}
-
-// A Source represents a remote repository in which vulnerability statements are stored
-type Source struct {
-	Repo   string `yaml:"repo"`
-	Branch string `yaml:"branch"`
-	Signed bool   `yaml:"signed"`
-	Rank   int    `yaml:"rank"`
-}
+import (
+	"io/ioutil"
+	"path/filepath"
+
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/rs/zerolog/log"
+	"github.com/sap/project-kb/kaybee/internal/errors"
+	"github.com/sap/project-kb/kaybee/internal/filesystem"
+	"gopkg.in/yaml.v2"
+)
 
 // A Policy determines how statements from different source repositories are
 // to be merged (reconciled, when conflicting)
-type Policy int
+type Policy string
 
 // Enumeration of the existing policy names
 const (
-	None Policy = iota
-	Strict
-	Soft
-	Latest
-	Oldest
+	None   Policy = "none"
+	Strict Policy = "strict"
+	Soft   Policy = "soft"
+	Latest Policy = "latest"
+	Oldest Policy = "oldest"
 )
 
-// PolicyFromString converts a policy name into a Policy instance
-func PolicyFromString(p string) Policy {
-	p = strings.ToLower(p)
-	switch p {
-	case "strict":
-		return Strict
-	case "soft":
-		return Soft
-	case "latest":
-		return Latest
-	case "oldest":
-		return Oldest
-	}
-	return None
-}
-
-func (p Policy) String() string {
-	switch p {
-	case Strict:
-		return "Strict"
-	case Soft:
-		return "Soft"
-	case Latest:
-		return "Latest"
-	case Oldest:
-		return "Oldest"
-	}
-	return "None"
+type Config interface {
```

Comment:
I understand we need to distinguish the Configuration struct from the Configuration interface, but the distinction is far from being self-evident.
The best practice is to use a noun that indicates what the interface (implementation) will do:
https://golang.org/doc/effective_go.html#interface-names

One option, could be ConfigurationReader or something like that.

Reply:
I think right now there are only:
```golang
type Config interface{}

// v1 implementation
type ConfigV1 struct{}

// v2 implementation
type ConfigV2 struct{}
```
and no `Configuration struct` left.

I think it's better if I rename this function for consistency:
```golang
func ParseConfiguration(file string) (Config, error) {}

// to
func ParseConfig(file string) (Config, error) {}
```

Reply:
Consistency is definitely a good idea :+1: 

However I very much prefer `Configuration` over `Config`.